### ISSUE TITLE
no-types-missing-file-annotation rule breaks if annotation not first line

### ISSUE THREAD

**Issue by zacanger** [gajus/eslint-plugin-flowtype#224]
If I have a file that has another comment on the first line, this rule errors.

Example:

```
/* eslint-disable no-useless-escape */
// @flow
```

Is this expected behavior?

**Comment by gajus** [comment]
cc @micaste

**Comment by micaste** [comment]
@gajus well that's how `isFlowFile` works currently (https://github.com/gajus/eslint-plugin-flowtype/blob/v2.32.1/src/utilities/isFlowFile.js#L17).
If you have the setting [onlyfileswithflowannotation](https://github.com/gajus/eslint-plugin-flowtype#onlyfileswithflowannotation) set to `true`, then all the rules will be ignored in the case mentioned by @zacanger.
That setting is not the 'recommended' one so it might be transparent for most people.

**Comment by micaste** [comment]
I guess that this use-case should be supported, to work as Flow itself:
- Supported:
```
// @flow
/**
 * @public
 */
type Param = string;
```
```
/**
 * @public
 */
// @flow
type Param = string;
```
```
/* eslint-disable */
// @flow
type Param = string;
```
- Not supported:
```
type Param = string;
// @flow
```

**Comment by mjomble** [comment]
Interestingly, the `require-valid-file-annotation` rule already allows cases where you have an ESLint comment before the annotation, etc.

I'd been using that rule for a while and all my code passed, but now that I updated to 2.33, I got a lot of failures from `no-types-missing-file-annotation` which seems to be using a different logic.

**Comment by NameFILIP** [comment]
also, if type definition is installed with `flow-typed` (for example, `flow-typed install camelcase@3`).
The file doesn't contain `@flow` comment, but instead it has:
```
// flow-typed signature: ff1dd5345a829c3f142d56256a01d7fb
// flow-typed version: <<STUB>>/styled-components_v2.0.0/flow_v0.45.0
```

**Comment by igidas** [comment]
Related with #235

**Comment by gajus** [comment]
This is a real problem when file needs to contain a shebang annotation, e.g.

This is valid, but will produce `flowtype/no-types-missing-file-annotation` warning.

```js
#!/usr/bin/env node

// @flow

```

This will result in a syntax error ("Parsing error: Unexpected character '#'").

```js
// @flow

#!/usr/bin/env node

```

**Comment by bobrovnik** [comment]
@gajus did you find any workaround for your issue?

**Comment by spudly** [comment]
Also a problem when telling jest to use jsdom in a test file because Jest requires its comment to be first, so the flow comment has to come after :(

```js
/**
 * @jest-environment jsdom
 */
// @flow
```

Swapping the two pragmas fixes the linter error, but breaks jest/jsdom.
